concordium-node/src/p2p/maintenance.rs

```diff
@@ -842,15 +853,37 @@ fn process_conn_change(node: &Arc<P2PNode>, conn_change: ConnChange) {
                 );
                 node.stats.soft_banned_peers.inc();
                 node.stats.soft_banned_peers_total.inc();
+                // If the peer was connected then also expunge it from
+                // the database so the node does not try to reconnect to it upon a restart.
+                if is_conn {
+                    if let Err(err) = node.remove_persisted_peer(remote_peer.addr) {
+                        warn!("Unable to remove 'StoredPeer' from database {}", err);
+                    }
+                }
             }
         }
         ConnChange::RemovalByToken(token) => {
             trace!("Removing connection with token {:?}", token);
-            node.remove_connection(token);
+            if let Some((is_conn, remote_peer)) = node.remove_connection(token) {
+                // If the peer was connected then also expunge it from
+                // the database so the node does not try to reconnect to it upon a restart.
+                if is_conn {
+                    if let Err(err) = node.remove_persisted_peer(remote_peer.addr) {
+                        warn!("Unable to remove 'StoredPeer' from database {}", err);
+                    }
+                }
+            }
         }
         ConnChange::RemoveAllByTokens(tokens) => {
             trace!("Removing connections with tokens {:?}", tokens);
-            node.remove_connections(&tokens);
+            let (_, removed_connected_peers) = node.remove_connections(&tokens);
+            for p in removed_connected_peers {
+                // If any connections were dropped, remove them
+                // from the database.
+                if let Err(err) = node.remove_persisted_peer(p.addr) {
+                    warn!("Unable to remove 'StoredPeer' from database {}", err);
```

Comment:
Shouldn't all this `warn` be `error`? They should not be happening unless something is off in configuration of the node.

Reply:
Yes, you're right.